pkg/storegateway/labelpb/types.pb.go

```diff
@@ -29,11 +30,10 @@ type Label struct {
 	Value string `protobuf:"bytes,2,opt,name=value,proto3" json:"value,omitempty"`
 }
 
-func (m *Label) Reset()         { *m = Label{} }
-func (m *Label) String() string { return proto.CompactTextString(m) }
-func (*Label) ProtoMessage()    {}
+func (m *Label) Reset()      { *m = Label{} }
+func (*Label) ProtoMessage() {}
```

Comment:
This `Label` type is identical to `LabelPair` here:
https://github.com/grafana/mimir/blob/c0a7b234d9ccbca237cd3f1fd2d3c41a96c22b55/pkg/mimirpb/mimir.pb.go#L239-L242

Mimir has `LabelAdapter` to do the zero-copy thing where Thanos has `ZLabel`.

(I would be ok with doing that change later, after this PR)

Reply:
I will look at it separately.

Reply:
I replaced ZLabel with LabelAdapter in #3345.